docs/standard/memory-and-spans/index.md

```diff
@@ -1,18 +1,20 @@
 ---
-description: "Learn more about: Memory- and span-related types"
-title: "Memory and spans"
-ms.date: "10/03/2018"
+description: "Learn more about: Memory-related and span-related types"
+title: "Memory-related and span types"
+ms.date: 07/12/2024
 helpviewer_keywords:
   - "Memory<T>"
   - "Span<T>"
   - buffers"
   - "pipeline processing"
 ---
-# Memory- and span-related types
+# Memory-related and span types
 
-Starting with .NET Core 2.1, .NET includes a number of interrelated types that represent a contiguous, strongly typed region of arbitrary memory. These include:
+.NET includes a number of interrelated types that represent a contiguous, strongly typed region of arbitrary memory. These types are designed to allow the creation of algorithms that *avoid copying memory or allocating on the managed heap* more than necessary. Creating them (either via `Slice` or their constructors) does not involve duplicating the underlying buffers: only the relevant references and offsets, which represent the "view" of the wrapped memory, are updated. In high-performance code, spans are often used to avoid allocating strings unnecessarily.
```

Comment:
`AsSpan` has the same effect as `Slice`.

Collection expressions are another way to create a span.